app/services/order_fees_handler.rb

```diff
@@ -1,12 +1,12 @@
 # frozen_string_literal: true
 
 class OrderFeesHandler
-  attr_reader :order, :distributor, :order_cycle
+  attr_reader :order
+
+  delegate :distributor, :order_cycle, to: :order
 
   def initialize(order)
     @order = order
-    @distributor = order.distributor
```

Comment:
I think it is not necessary to store a reference to the `order_cycle` and the `distributor` if we can delegate those two to the order.

Also, it was a source of problems when I was writing the tests.
 
```ruby
create(:simple_order_cycle)
create(:order_with_distributor, distributor: distributor)  # order created fees_handler.order_cycle = nil
order.update!({order_cycle_id: order_cycle.id,ship_address_id: ship_address.id})
order.reload
order.recreate_all_fees! # this will not work, because the OrderFeesHandler was initialized with  order_cycle= nil

```